Basic_Python_Diploma/YaDisk.py

```diff
@@ -0,0 +1,424 @@
+            print()
+        else:
+            print(test)
+
+        self.reload()
+        print("Текущий список папок:")
+        self.print_all('folder')
+
+    def find_biggest(self, obj_type=None):
+        """For find_biggest you shall indicate 'obj_type'-argument: either 'file' or 'folder' """
+        lst = None
+        filename = None
+        if (obj_type is None) or (obj_type != 'file' and obj_type != 'folder'):
+            return print(self.find_biggest.__doc__)
+        else:
+            if obj_type == 'file':
+                lst = self.all_files
+                filename = "biggest_file_info.json"
+            elif obj_type == 'folder':
+                lst = self.all_folders
+                filename = "biggest_folder_info.json"
+        max_size = 0
+        for obj in lst:
+            if obj.size > max_size:
+                max_size = obj.size
+        for obj in lst:
+            if max_size == obj.size:
+                with open(filename, "w", encoding="UTF-8") as f:
+                    json.dump({obj.name: obj.size}, f)
+                return obj
+
+    def delete(self, *objects):
+        """метод удаляет папку или файл с яндекс.диска в корзину или навсегда"""
+
+        print("\nДля удаления объекта(-ов) в Корзину просто нажмите Enter.\n"
+              "Для полного удаления объекта(-ов) без возможности восстановления введите 1.")
+        permanent = input("Введите ответ: ")
+        perm_del = ''
+        if permanent == "1":
+            perm_del = {"permanently": "true"}
+
+        for obj in objects:
+            param = {'path': obj.object_realpath}
+            if perm_del:
+                param.update(perm_del)
+            put = r.delete(self.URL, headers=self.headers, params=param)
+            if not str(put.status_code).startswith('2'):
+                return put.status_code, put.json()
+            else:
+                self.reload()
+
+        for num, dir in enumerate(ya.all_folders):
+            print("dir" + str(num) + ".", dir)
+        string = ''
+        if len(objects) == 1:
+            string = objects[0]
+        elif len(objects) > 1:
+            string = f'{", ".join(objects)}'
+
+        if perm_del:
+            return f'Объекты {string} успешно удалены с Яндекс.Диска.'
+        else:
+            return f'Объекты {string} успешно удалены в Корзину.'
+
+    def download(self, item):
+        target_folder = 'downloads'
+        if os.path.exists(target_folder) is False:
+            os.mkdir(target_folder)
+        try:
+            if item.type == 'dir':
+                target_folder = os.path.join(target_folder, item.object_realpath.split('disk:/')[-1])
+                if os.path.exists(target_folder) is False:
+                    os.mkdir(target_folder)
+                    target_folder = os.path.abspath(target_folder)
+                else:
+                    target_folder = os.path.abspath(target_folder)
```

Comment:
Перефразируем мысль:

```suggestion
                if not os.path.exists(target_folder):
                    os.mkdir(target_folder)

                target_folder = os.path.abspath(target_folder)
```